ui/src/actions/sampleview.js

```diff
@@ -143,7 +143,21 @@ export function setGridOverlay(level) {
 }
 
 export function toggleDrawGrid() {
-  return { type: 'DRAW_GRID' };
+  return async (dispatch, getState) => {
+    const { sampleview, queue } = getState();
+
+    // Stop three-click centring if active
+    if (sampleview.clickCentring) {
+      await dispatch(abortCentring());
+    }
+
+    if (!sampleview.drawGrid && !queue.currentSampleID) {
+      dispatch(showErrorPanel(true, 'There is no sample mounted'));
+      return;
+    }
```

Comment:
Note that it is now allowed to toggle off `drawGrid` when there is no sample mounted.